book/further-information/04-pip-installing/why/main.md

```diff
@@ -0,0 +1,116 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.12
+    jupytext_version: 1.6.0
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Further information.
+
+## Why do we do `python -m pip install` and not `pip install`
+
+In some places you will see that to install a python library you can run the
+following in the command line:
+
+```bash
+$ pip install <library>
+```
+
+Instead of:
+
+```bash
+$ python -m pip install <library>
+```
+
+This will work and most often will have the same effect however it is not
+recommended.
+
+On some occasions `pip`, which is the library used to install packages will not
+correspond to the same `python` install you are using. To ensure this does not
+create any problems it is recommended to use `python -m pip install <library>`.
+
+Here is a short blog post discussing this:
+<https://adamj.eu/tech/2020/02/25/use-python-m-pip-everywhere/>.
+
+## How to install multiple libraries at a time
+
+If you want to keep a record of multiple python libraries that you want to
+install, the convention is to keep them in a file called `requirements.txt`.
+
+Then to install all the libraries you can run the following in the command line:
+
+```bash
+$ python -m pip install -r requirements.txt
+```
+
+For example, the `requirements.txt` file for this software used to write this
+book looks like:
+
+```
+black==20.8b1
+interrogate==1.3.1
+invoke==1.4.1
+isort==4.3.21
+jupyter-book==0.8.1
+jupytext==1.6.0
+nbval==0.9.6
+proselint==0.10.2
+pytest==5.4.3
+```
+
+If the version specifications are omitted then the latest versions of the
+libraries would be installed.
+
+```
+black
+interrogate
+invoke
+isort
+jupyter-book
+jupytext
+nbval
+proselint
+pytest
+```
+
+## What is the Python package index
+
+The Python package index commonly referred to as PyPi is:
```

Comment:
```suggestion
The Python package index commonly referred to as PyPI is:
```

Is that right?

Reply:
You use `PyPI` in line 87 below.

Reply:
I'm so inconsistent all over the place. At some point I need to go back through everything for that.